### DISCUSSION TITLE
Data importer: Populating a field collection field

### DISCUSSION THREAD

**Discussion by sissonen** [pimcore/pimcore#9846] [Q&A / Support]
I have a Product class which has a field collection field for providers. The field collection consists of three fields: a Many-to-One relation to another DataObject class (Provider), a textfield (provider's product code) and a number field (provider's price).

I have a CSV file which has the provider data listed: product code to link the Product and Provider id (as well as the price and provider's product code).

With Pimcore 6.8 this field collection creation was quite easy to do with CSV import + PHP code -operator, but I suppose a PHP code operator is not available with the current data importer implementation.

Is there a way to create field collections with data importer? Is there a plan to implement something for this?

**Comment by fashxp** [comment]
Hi,
you would need to create a custom data target implementation ... that probably contains similar logic as your php code operator. 
maybe this also would be something of general interest and a great contribution to the data-importer? 

cheers...

**Reply by sissonen** [reply]
With my current work situation I, unfortunately, don't have time to start implementing this. I went with a controller based solution outside of the data-importer.

**Reply by FelixScalpers** [reply]
Hello,
im just starting to learn and use PIMCORE, and after checking the posibility of importing some fields from CSV to a specific field-collection i came to this post, so far i have realize that im not able to export a Excel colum into one of the field collections within my data Object, and i guess i need to write my own data target implementation but i dont really know where to start. i mean if you could give me any advice it will be welcome.
Thanks in advance
